testdata/p4_14_samples_outputs/issue1013.p4-stderr

```diff
@@ -0,0 +1,3 @@
+issue1013.p4(13): [--Wwarn=mismatch] warning: -0x1: negative value with unsigned type
+    modify_field(h.x, h.x, 0x0);
```

Comment:
Something is wrong here. There are a few other strange error messages.

Reply:
Strange indeed. Investigating this further, this is caused by the (strange?) 14-to-16 translation. In particular, the `modify_field` is translated as below for some reason. That said, I think this PR didn't cause this strange error message but simply revealed it.

```
771: ./tmp6u3htctk/issue1013.p4(24): [--Wwarn=mismatch] warning: -0x1: negative value with unsigned type
771:         hdr.h.x = hdr.h.x & ~32w0x0 | hdr.h.x & 32w0x0;
771:                              ^^^^^^
```

Reply:
The translation looks ok -- this is a modify field with mask, so it becomes `dest = dest & ~mask | src & mask`; setting just the masked bits of the destination, leaving other bits unchanged.  The weirdness come from why it thinks `~32w0x0` is a negative constant -- it is not.  It should be `32w0xffff` after constant folding.  Not sure where the -1 comes from.

Reply:
This seems to be simply caused by the comparison `< 0` with `value` of type `bit_int` which is `boost::multiprecision::mpz_int` or `cpp_int` which treats all values as signed?

https://github.com/p4lang/p4c/blob/90a8f3f7cbd242c8759f66cb7cf8e3966a173e18/ir/expression.cpp#L94